### ISSUE TITLE
Model might be overfitted and mAP is not that good

### ISSUE THREAD

**Issue by B1D1ng** [AlexeyAB/darknet#6650]
* what command do you use?
./darknet detector train data/obj.data cfg/yolov3-tiny.cfg yolov3-tiny.conv.11 -map

    * what dataset do you use?   
fire and smoke images I download from Google and some open dataset

    * what Loss and mAP did you get?
    * show chart.png with Loss and mAP    

![chart](https://user-images.githubusercontent.com/31911710/92917699-4ed7ec00-f461-11ea-9e39-fec3d889fd61.png)

   My dataset includes 30k images(13k positive samples and 17k negative samples), In cfg file I set the max_batches=100000 but I stop training when the iteration is 50000 because I think the model might be overfitted. As the chart.png above we can see that the mAP of test set  fluctuates around 66%, when I set the valid=./train.txt and calculate the mAP of train set, it reach around 80%. So I think the model might be overfitted. 
  In addition, the final loss is around 0.38 and I think it is very low but the mAP of the test set is not that good, I think the model I train would not work well when I use it to test new images or videos out of my dataset..

So, may I ask some questions:
1. How can I stop training early so as to avoid overfitting, now I just calculate the train mAP of the weights that I got from backup folder and compare to the chart.png, finally select the best weights which is not overfitted.
2. How can I improve the mAP of test set? Compare the train mAP curve and test mAP curve I fould that if I want to avoid overfitting, I should end training early. However, the mAP of the test set might be not that high. 

Hope to your help, thank you!

**Comment by LukeAI** [comment]
I think you should train for a longer time, it might stabilise - we can't tell yet. You should probably use a lower learning rate until the curve smoothes and maybe lower subdivisions if you have the VRAM. This is not overfitted - this is under-trained.

You can stop training early just by cancelling the training process.

**Comment by B1D1ng** [comment]
> I think you should train for a longer time, it might stabilise - we can't tell yet. You should probably use a lower learning rate until the curve smoothes and maybe lower subdivisions if you have the VRAM. This is not overfitted - this is under-trained.
> 
> You can stop training early just by cancelling the training process.

The test mAP curve and train mAP curve is as follows:
![mAP](https://user-images.githubusercontent.com/31911710/93054841-95655a80-f69c-11ea-92ce-60b65cf1a603.png)
And after 50000 Iterations, I set the learning rate to 0.00001 and continute training 40000 iterations, finally the test mAP curve is as follows:
![92000](https://user-images.githubusercontent.com/31911710/93054968-c9408000-f69c-11ea-888d-39b0cd9d964e.png)

How can I improve the test mAP next?

**Comment by UsefNait** [comment]
hello @B1D1ng ,how could you plot map and loss in this way plzzz?

**Comment by RamaRam99** [comment]
> > I think you should train for a longer time, it might stabilise - we can't tell yet. You should probably use a lower learning rate until the curve smoothes and maybe lower subdivisions if you have the VRAM. This is not overfitted - this is under-trained.
> > You can stop training early just by cancelling the training process.
> 
> The test mAP curve and train mAP curve is as follows:
> ![mAP](https://user-images.githubusercontent.com/31911710/93054841-95655a80-f69c-11ea-92ce-60b65cf1a603.png)
> And after 50000 Iterations, I set the learning rate to 0.00001 and continute training 40000 iterations, finally the test mAP curve is as follows:
> ![92000](https://user-images.githubusercontent.com/31911710/93054968-c9408000-f69c-11ea-888d-39b0cd9d964e.png)
> 
> How can I improve the test mAP next?

How were you able to plot that? sorry, im a novice.
